### PR TITLE
Add PBF Setting to Engine Preferences

### PR DESCRIPTION
**Pull request by RoiEXLab**
## Overview
This PR is the last preparation before being able to remove the IBean logic to store data completely.

#### What Game specific information to store
Store the name of a dice roller in the Game Properties to be able to look it up when loading the game.
Store From/To Email addresses for Dice server in the Game Properties.
Store Game-ID (if present) of a dice roller in the Game Properties.

Store Forum Name in the GameProperties to be able to look it up when loading the game.
Store Forum Topic ID form PbF in the Game Properties.
Store Forum attach save game to summary boolean.

Store PbEM Subject in the Game Properties
Store PbEM Opponent in the Game Properties

Store "Also post after combat move" in the Game Properties.

#### Things to look out for
The game still needs to be able to "get" a current instance of a dice roller without it being part of the Game Properties, not quite sure how the internal logic works there.

## Functional Changes
A do-nothing entry has been added to engine preferences.

## Manual Testing Performed
Well, changes are getting saved.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java [hunk 1]
```diff
@@ -568,6 +568,54 @@ public void reset() {
...
+            password -> context.setValue(
+                passwordSetting,
+                (password.length == 0) ? null : password,
+                SaveContext.ValueSensitivity.SENSITIVE));
+      }
+
+      @Override
+      public void resetToDefault() {
+        usernameField.setText(credentialToString(usernameSetting::getValue));
+        passwordField.setText(credentialToString(passwordSetting::getValue));
```
[the remark this answers is not in the document]
Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java]: Yes, my bad 😅

File: game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java [hunk 2]
```diff
@@ -568,6 +568,54 @@ public void reset() {
     };
   }
 
+  static SelectionComponent<JComponent> forumPosterSettings(
+      final ClientSetting<char[]> usernameSetting,
+      final ClientSetting<char[]> passwordSetting
+  ) {
```
[the remark this answers is not in the document]
Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java]: Changed this to match the formatter.
Not really familiar with formatting configurations, but I  agree that the "symmetric" way looks better.

**Comment by panther2** [comment]
@RoiEXLab @ssoloff 
Just to be sure: The settings are saved encrypted?

**Comment by ssoloff** [comment]
> There are some limitations though: If no master password is set (I don't even know how to set one tbh, but I think it should be possible somehow?) an "empty key" is used to encrypt everything which has to be stored somewhere for the engine to decrypt the password when it requires it.

@RoiEXLab The master password is automatically generated, if not already present, with approximately 160 bits of entropy.  Currently, it's an opaque value that the user never sees nor manipulates.  But the system is set up in such a way that we could introduce the ability for a user to enter their own master password so that it is never persisted at all.  (The master password is currently stored in the file system with permissions that only allow the owner to read/write it.)

> The settings are saved encrypted?

@panther2 Just to be precise, the email/forum usernames and passwords are encrypted but nothing else is (e.g. hostname and port of your email provider).  This is consistent with what you and I discussed in #1884.

**Comment by panther2** [comment]
@RoiEXLab @ssoloff Perfect, thank you for the explanations.

**Pull request merged**